test/e2e/controller_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ViaQ/log-exploration-api/pkg/configuration"
+	logscontroller "github.com/ViaQ/log-exploration-api/pkg/controllers/logs"
+	"github.com/ViaQ/log-exploration-api/pkg/elastic"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var logsController *logscontroller.LogsController
+var router *gin.Engine
+var w *httptest.ResponseRecorder
+
+func TestMain(t *testing.T) {
+	log, _ := initCustomZapLogger("info")
+	appConf := configuration.ParseArgs()
+	logProvider, _ := elastic.NewElasticRepository(log.Named("elasticsearch"), appConf.Elasticsearch)
+	router = gin.Default()
+	logsController = logscontroller.NewLogsController(log, logProvider, router)
+	w = httptest.NewRecorder()
+}
+
+func TestGetAllLogs(t *testing.T) {
+	req, _ := http.NewRequest("GET", "/logs/", nil)
+	status := w.Code
+	router.ServeHTTP(w, req)
+	p, err := ioutil.ReadAll(w.Body)
+	pageOK := err == nil && strings.Index(string(p), "Logs") > 0
+
+	if status == http.StatusBadRequest {
```

Comment:
nit: Consider using `switch/case` construction to make the code more clear
````
switch status {
case http.StatusBadRequest, StatusInternalServerError:
         ....
         t.Errorf("Failed with status: %s", status)
case http.StatusOk:
         ....
}
````

Reply:
Updated the following changes. But one issue I'm still facing is I have created `TestMain` in both the test files and now it is giving error for `multiple declaration of TestMain`. I initially thought to resolve this using `subtests` but I don't think that would be a good way to do it and it would resolve things completely.